third_party/move/tools/move-package/src/compilation/compiled_package.rs

```diff
@@ -748,21 +745,19 @@ impl CompiledPackage {
             if skip_attribute_checks {
                 flags = flags.set_skip_attribute_checks(true)
             }
-
-            let model = match (
-                resolution_graph.build_options.generate_docs,
-                resolution_graph.build_options.generate_abis,
-                optional_global_env,
-            ) {
-                (false, false, Some(env)) => env, // Use V2 generated model if not for docgen or abigen
-                _ => run_model_builder_with_options_and_compilation_flags(
-                    // Otherwise, use V1 generated model
-                    vec![sources_package_paths],
-                    deps_package_paths.into_iter().map(|(p, _)| p).collect_vec(),
+            let model = if let Some(env) = optional_global_env {
```

Comment:
With this change, if you run e2e move test with V2 flag, the doc will be different from what V1 generates.

Reply:
I don't see it.  Can you please tell me a command line to show this?

To be honest, I'm surprised this works.  I had to undo a similar thing in the previous PR to get it to land.  But I guess now things are somehow happier.  We may be really running V1 under the covers here; I haven't tried enabling `MOVE_COMPILER_BLOCK_V1` recently.

Reply:
You could try to run `MOVE_COMPILER_V2=true cargo test` in e2e-move-tests or compare the result of `aptos move document` with V1 and V2

Reply:
e2e-move-tests doesn't generate any docs.  I've seen a diff with `aptos move document` but it was a pain to set up.  We desperately need some aptos move cli tests.  (Or tests of the functionality.)

Reply:
I also was able to see a difference by installing the aptos cli and running it, but that's quite a pain.  Do we have aptos move cli test cases somewhere?

Reply:
I only found the smoke test for aptos cli: https://github.com/aptos-labs/aptos-core/tree/main/testsuite/smoke-test/src/aptos_cli